crates/red_knot_python_semantic/src/types.rs

```diff
@@ -343,6 +343,10 @@ fn symbol_from_bindings<'db>(
              visibility_constraint,
          }| {
             let binding = binding?;
+            if binding.in_stub(db) && !binding.is_reexported(db) {
+                return None;
+            }
+
```

Comment:
We need to know here whether the binding is in the same file as the caller i.e., the querying file. To do that we'd need to pass this information via a ton of functions:
* `symbol_from_bindings`
* `symbol_from_declaration`
* `symbol`
* `global_symbol`
* `builtins_symbol`
* `known_module_symbol`
* `typing_symbol`
* `typing_extensions_symbol`
* `Type::member`
* `ModuleLiteralType::member`

Why `member` methods? The `from builtins import Iterable` call will try to find the symbol via `ModuleLiteralType::member` which is one of the main point where we need to pass this information.